public_html/wp-content/plugins/pattern-creator/includes/openverse-client.php

```diff
@@ -0,0 +1,207 @@
+				'grant_type' => 'client_credentials',
+			),
+		) );
+		if ( is_wp_error( $response ) ) {
+			return $response;
+		}
+
+		$response_code = wp_remote_retrieve_response_code( $response );
+		if ( 200 !== $response_code ) {
+			return new WP_Error(
+				'invalid-ov-token-request',
+				sprintf( __( 'The token generation request failed with a %s error.', 'wporg-patterns' ), $response_code )
+			);
+		}
+
+		$body = wp_remote_retrieve_body( $response );
+		if ( is_wp_error( $body ) ) {
+			return $body;
+		}
+
+		$token = json_decode( $body, true ); // Second param returns decoded value as assoc array.
+		if ( null === $token ) {
+			return new WP_Error( 'invalid-ov-token-response', __( 'The token generation response is malformed.', 'wporg-patterns' ) );
+		}
+
+		// Invalidate the token 5 minutes early, just in case.
+		$token['expires_at'] = time() + $token['expires_in'] - 5 * MINUTE_IN_SECONDS;
+		update_option( self::TOKEN_OPTION_KEY, $token );
+
+		return $token['access_token'];
+	}
+
+	/**
+	 * Get the results from the Openverse API.
+	 *
+	 * @return WP_Error|array
+	 */
+	public function fetch_results() {
+		$auth_token = $this->get_oauth_token();
+		if ( is_wp_error( $auth_token ) ) {
+			return $auth_token;
+		}
+
+		$url = add_query_arg(
+			array(
+				'format' => 'json',
+				'license' => 'cc0',
+				'q' => $this->get_param( 'search' ),
+				'page' => $this->get_param( 'page' ),
+				'page_size' => $this->get_param( 'per_page' ),
+			),
+			$this->url . '/v1/images',
+		);
+		$response = wp_remote_get(
+			$url,
+			array(
+				'headers' => array(
+					'Authorization' => 'Bearer ' . $auth_token,
+				),
+			)
+		);
+		if ( is_wp_error( $response ) ) {
+			return $response;
+		}
+
+		$response_code = wp_remote_retrieve_response_code( $response );
+		if ( 200 !== $response_code ) {
+			return new WP_Error(
+				'invalid-openverse-request',
+				sprintf( __( 'The Openverse API request failed with a %s error.', 'wporg-patterns' ), $response_code )
+			);
+		}
+
+		$body = wp_remote_retrieve_body( $response );
+		if ( is_wp_error( $body ) ) {
+			return $body;
+		}
+
+		$data = json_decode( $body );
+		if ( null === $data ) {
+			return new WP_Error( 'invalid-openverse-response', __( 'The Openverse API response is malformed.', 'wporg-patterns' ) );
+		}
+
+		return $data;
+	}
+
+	/**
+	 * Return the list of search results, possibly from a local cache.
+	 *
+	 * @return WP_Error|array
+	 */
+	public function search() {
+		if ( ! $this->get_param( 'search' ) ) {
+			return new WP_Error( 'invalid-empty-search', __( 'Search term cannot be empty.', 'wporg-patterns' ) );
+		}
+
+		$cache_key = self::CACHE_KEY . md5( wp_json_encode( $this->get_params() ) );
+		$ttl = HOUR_IN_SECONDS;
+
+		$results = get_transient( $cache_key );
+		if ( false === $results ) {
+			$results = $this->fetch_results();
+			if ( is_wp_error( $results ) ) {
+				return $results;
+			}
+			set_transient( $cache_key, $results, $ttl );
+		}
```

Comment:
```suggestion
		$results = get_transient( $cache_key );
		if ( false === $results ) {
			$results = $this->fetch_results();
			if ( is_wp_error( $results ) ) {
				// During development, return the upstream error.
				if ( defined( 'WP_DEBUG' ) && WP_DEBUG ) {
					return $results;
				}
			
				// Log the error
				trigger_error( $results->get_error_code(), E_USER_WARNING );

				// Set a short timeout to avoid hammering the API during outages.
				set_transient( $cache_key, [], 0.5 * MINUTE_IN_SECONDS );
		
				return new WP_Error(
					'search-request-failed',
					__( 'The search API provider is currently experiencing an error. Try again in a few seconds', 'wporg-patterns' ) // Publicly facing human-readable error.
				);
			}
			set_transient( $cache_key, $results, $ttl );
		}
```